vllm/sequence.py

```diff
@@ -973,6 +998,16 @@ def prompt_adapter_num_virtual_tokens(self) -> int:
         return self.prompt_adapter_request.prompt_adapter_num_virtual_tokens \
                         if self.prompt_adapter_request else 0
 
+    # Multi-Step Chunked-Prefill property
+    @property
+    def is_single_step_prompt(self) -> bool:
+        return self.is_prompt and self.do_sample
```

Comment:
Why `self.is_prompt and self.do_sample` means "single step prompt"?

Reply:
do_sample is true, only when the token_chunk_size matches the num_uncomputed_tokens of the sequence. This indicates that the prompt will finish processing in a single `execute_model` step.

I have added a comment explaining the same.